pkg/ccl/backupccl/backup_test.go

```diff
@@ -10125,3 +10144,82 @@ func TestBackupRestoreSeparateExplicitIsDefault(t *testing.T) {
 		sqlDB.Exec(t, "DROP DATABASE inc_fkdb;")
 	}
 }
+
```

Comment:
general test comment: can we come up with a minimal unit test for the method that generates timestamped files. Maybe a subtest for each provider that writes a fixed number of timestamped files, lists them with limit 1, and then ensures that our lexicographic sorting algorithm is sound?

Reply:
Didn't get a chance on friday to make buckets for gcs and azure and actually test it for those two(but definitely will), but the general test is there.